src/cdi_ndi/test/tstNDI_TN.cc

```diff
@@ -0,0 +1,139 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   cdi_ndi/test/tstNDI_TN.cc
+ * \author Ben R. Ryan
+ * \date   2019 Nov 18
+ * \brief  NDI_Base test
+ * \note   Copyright (C) 2019-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+#include "cdi_ndi/NDI_Common.hh"
+#include "cdi_ndi/NDI_TN.hh"
+#include "cdi/CDI.hh"
+#include "ds++/Release.hh"
+#include "ds++/ScalarUnitTest.hh"
+#include "ds++/dbc.hh"
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+using rtt_cdi_ndi::NDI_TN;
+using rtt_dsxx::soft_equiv;
+
+//----------------------------------------------------------------------------//
+// TESTS
+//----------------------------------------------------------------------------//
+
+void gendir_test(rtt_dsxx::UnitTest &ut) {
+
+  // Write a custom gendir file to deal with NDI-required absolute path to data
+  std::string gendir_in = "gendir_tmp.all";
+  std::string gendir_tmp_path = ut.getTestInputPath() + gendir_in;
+  std::string data_path = ut.getTestSourcePath() + "ndi_data";
+  std::ofstream gendir_tmp_file;
+  gendir_tmp_file.open(gendir_tmp_path);
+  gendir_tmp_file << "tndata\n";
+  gendir_tmp_file << "  z=n+be7->p+li7.040ztn  d=12/20/2004  l=lanl04\n";
+  gendir_tmp_file << "    f=" << data_path << "\n";
+  gendir_tmp_file << "    ft=asc  ln=73  o=3372  end\n";
+  gendir_tmp_file.close();
+
+  std::string gendir_path = gendir_tmp_path;
+  std::string library_in = "lanl04";
+  std::string reaction_in = "n+be7->p+li7";
+
+  NDI_TN tn(gendir_path, library_in, reaction_in,
+            rtt_cdi_ndi::DISCRETIZATION::MULTIGROUP);
+
+  // Check return values of getters
+  FAIL_IF(tn.get_gendir().find(gendir_in) == std::string::npos);
+  FAIL_IF_NOT(tn.get_dataset() == "tn");
+  FAIL_IF_NOT(tn.get_library() == "lanl04");
+  FAIL_IF_NOT(tn.get_reaction() == "n+be7->p+li7");
+  FAIL_IF_NOT(tn.get_reaction_name() == "n+be7->p+li7.040ztn");
+  FAIL_IF_NOT(tn.get_discretization() ==
+              rtt_cdi_ndi::DISCRETIZATION::MULTIGROUP);
+  FAIL_IF_NOT(tn.get_num_products() == 2);
+  auto products = tn.get_products();
+  FAIL_IF_NOT(products.size() == 2 && products[0] == 1001 &&
+              products[1] == 3007);
+  auto multiplicities = tn.get_product_multiplicities();
+  FAIL_IF_NOT(multiplicities.size() == 2 && multiplicities[0] == 1 &&
+              multiplicities[1] == 1);
+  auto reaction_temperature = tn.get_reaction_temperature();
+  FAIL_IF_NOT(reaction_temperature.size() == 3);
+  FAIL_IF_NOT(soft_equiv(reaction_temperature.front(), 1.000000000000000056e-01,
```

Comment:
Since you are only checking 8 digits, it seems weird that you provide 16 digits for the gold value.  Not really an issue.

Reply:
I just didn't want to check the same number of digits as I provided. Only checking 8 digits is probably overly cautious, if there's an established style for calls like this I can change these. Also, I normalized some very small values to avoid `soft_equiv` not catching differences near 0 in an absolute sense.